docs/reference/aggregations/bucket/frequent-items-aggregation.asciidoc

```diff
@@ -0,0 +1,159 @@
+to discover relationships between different data points (items) in a data set 
+and to create association rules based on those relationships.
+
+
+==== Syntax
+
+A `frequent_items` aggregation looks like this in isolation:
+
+[source,js]
+--------------------------------------------------
+"frequent_items": {
+  "minimum_set_size": 3,
+  "fields": [
+    {"field": "my_field_1"},
+    {"field": "my_field_2"}
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+.`frequent_items` Parameters
+|===
+|Parameter Name |Description |Required |Default Value
+|`field` |(array) Fields to analyze. | Required |
+|`minimum_set_size` | (integer) The minimum size of one item set. | Optional | `1`
+|`minimum_support` | (integer) The minimum support of one item set. | Optional | `0.1`
+|`size` | (integer) The number of top item sets to return. | Optional |
+|===
+
+
+[discrete]
+[[frequent-items-minimum-support]]
+==== Minimum support
+
+Support represents the frequency of an item set normalized between 0 
+and 1. It is calculated by dividing the number of documents containing the item 
+set by the total number of documents.
+
+For example, if a given item set is contained by five documents and the total 
+number of documents is 20, then the support of the item set is 5/20 = 0.25. 
+Therefore, this set is returned only if the minimum support is 0.25 or lower. As 
+a lower minimum support prunes more items, the calculation is less resource 
+intensive. The `minimum_support` parameter has an effect on the required memory 
+and the runtime of the aggregation.
+
+
+[discrete]
+[[frequent-items-minimum-set-size]]
+==== Minimum set size
+
+The minimum set size is the minimum number of item the set needs to contain. A 
+value of 1 returns the frequency of single items. The higher the minimum set 
+size the less items are returned. For example, the item set `cat, dog, 
+crocodile` is only returned if the minimum set size is 3 or lower.
+
+
+[discrete]
+[[frequent-items-size]]
+==== Size
+
+This parameter defines the maximum number of item sets to return. The result 
+contains top-k item sets; the item sets with the highest support values. This 
+parameter has a significant effect on the required memory and the runtime of the 
+aggregation.
+
+
+[discrete]
+[[frequent-items-example]]
+==== Example
+
+In the following example, we use the e-commerce {kib} sample data set. The goal 
+is to find out from what product categories the customers purchase products 
+frequently together based on transaction data. We are interested in sets with 
+two items, and want to see the first three frequent item setes with the highest 
+support.
+
+[source,console]
+-------------------------------------------------
+GET kibana_sample_data_ecommerce /_search 
+{
+  "size": 0,
+  "aggs": {
+    "my_agg": {
+      "frequent_items": {
+        "minimum_set_size": 2,
+        "fields": [
+          {
+            "field": "category.keyword"
+          }
```

Comment:
I would really like an example with at least 2 fields, can't we use one of the geoip fields in addition?

Reply:
Sure, I added it via [2a7e3fd](https://github.com/elastic/elasticsearch/pull/86037/commits/2a7e3fdd7377d14483e4a794c56b69eb78b04353).